src/view/UserphraseView.h

```diff
@@ -64,4 +64,7 @@ public slots:
     UserphraseViewMenu *menu_;
 
     bool dialogType_;
+
+    std::shared_ptr<QString> originalPhrase;
```

Comment:
for now, it's ok to use `std::shared_ptr<QString>` :\

-

The reason of using `std::shared_ptr<QString>` in this case is flawed IMO:
`QString` already has the nature of ["implict sharing"](http://doc.qt.io/qt-5/implicit-sharing.html) which means it doesn't need `std::shared_ptr` to help maximize resource usage and minimize copying.
This should be refactored in another PR....